Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -882,6 +882,15 @@ void StyleComputer::cascade_declarations(
     auto cascade_style_declaration = [&](CSSStyleProperties const& declaration) {
         seen_properties.fill(false);
         for (auto const& property : declaration.properties()) {
+
+            // OPTIMIZATION: If we're computing the logical alias mapping context, we only care about the properties
+            //               that affect the logical mappings.
+            if (!logical_alias_mapping_context.has_value()) {
+                if (property.property_id != PropertyID::WritingMode && property.property_id != PropertyID::Direction) {
```

Comment:
`text-direction` is also needed, once we implement that. Maybe add a FIXME about it?

Reply:
Even better!